### PR TITLE
show specific err msg on password form

### PR DESCRIPTION
**Pull request by thesahindia**
### Details

Proposal: https://github.com/Expensify/App/issues/4856#issuecomment-906656039
Approval : https://github.com/Expensify/App/issues/4856#issuecomment-908557025
As discussed, I've added conditions for 3 specific error messages on password form.

### Fixed Issues

$  https://github.com/Expensify/App/issues/4856

### QA Steps

1. Launch the app
2. Enter email or phone number & proceed
3. Click on signin without entering password

### Tested On

- [x] Web
- [x] Mobile Web
- [ ] Desktop
- [ ] iOS
- [x] Android

#### Web

web

#### Mobile Web

Mobile web

#### Android

android

### CODE REVIEW

File: src/pages/signin/PasswordForm.js [hunk 1]
```diff
@@ -57,12 +57,19 @@ class PasswordForm extends React.Component {
...
     validateAndSubmitForm() {
         if (!this.state.password.trim()
-            || (this.props.account.requiresTwoFactorAuth && !this.state.twoFactorAuthCode.trim())
+            && (this.props.account.requiresTwoFactorAuth && !this.state.twoFactorAuthCode.trim())
         ) {
             this.setState({formError: 'passwordForm.pleaseFillOutAllFields'});
             return;
         }
-
+        if (!this.state.password.trim()) {
```
Comment (puneetlath) [inline src/pages/signin/PasswordForm.js:65]: Not strictly required by our style guide, but I think it'd be cleaner if there was a line break between each conditional. That seems to be how it is more frequently done in our codebase from what I can see.

File: src/pages/signin/PasswordForm.js [hunk 2]
```diff
@@ -57,12 +57,19 @@ class PasswordForm extends React.Component {
      */
     validateAndSubmitForm() {
         if (!this.state.password.trim()
-            || (this.props.account.requiresTwoFactorAuth && !this.state.twoFactorAuthCode.trim())
+            && (this.props.account.requiresTwoFactorAuth && !this.state.twoFactorAuthCode.trim())
```
Comment (puneetlath) [inline src/pages/signin/PasswordForm.js:60]: Doesn't change the result, but the parenthesis around `this.props.account.requiresTwoFactorAuth && !this.state.twoFactorAuthCode.trim()` aren't necessary.

**Review by puneetlath — COMMENTED** [review]
A couple minor comments. Up to you whether you agree or not. 

Also, just waiting for a response on this: https://github.com/Expensify/App/issues/4856#issuecomment-910459210

**Comment by thesahindia** [comment]
Making the required changes!

File: src/pages/signin/PasswordForm.js (hunk 1 above)
Comment (thesahindia) [reply src/pages/signin/PasswordForm.js:64]: Added line break after each `if statement`

File: src/pages/signin/PasswordForm.js (hunk 2 above)
Comment (thesahindia) [reply src/pages/signin/PasswordForm.js]: Removed the unnecessary brackets and also moved the conditions into single line.

Comment (puneetlath) [reply src/pages/signin/PasswordForm.js]: Awesome, thanks!

File: src/pages/signin/PasswordForm.js (hunk 1 above)
Comment (puneetlath) [reply src/pages/signin/PasswordForm.js:64]: Sweet, thanks.

**Review by puneetlath — APPROVED** [review]
Looks great!

**Comment by botify** [comment]
This has been deployed to production and is now subject to a 7-day regression period.
If no regressions arise, payment will be issued on 2021-09-10.  :confetti_ball:

**Pull request merged**